radio/src/gui/480x272/lcd.cpp

```diff
@@ -557,3 +557,272 @@ void DMABitmapConvert(uint16_t * dest, const uint8_t * src, uint16_t w, uint16_t
   }
 }
 #endif
+
+//OW
+//THANKS to Adafruit and its GFX library !
+//https://learn.adafruit.com/adafruit-gfx-graphics-library
+
+void lcdDrawCircleQuarter(coord_t x0, coord_t y0, int16_t r, uint8_t corners, LcdFlags att)
```

Comment:
what about using the function in libopenui to draw an arc with any angle?

Reply:
this is an interesting point (and one of the pain points I indicated): I certainly have noted that there are some arc/circle functions. However, their status was totally unclear to me. They are not used anywhere and not exposed to lua. For some also only the function prototypes seem to exist but not the implementation. So, I simply could not make sense of what I saw, and thus added what I had (the reason I added to the C code and didn't do it in the lua script is that I found that the execution rate of my lua script did increase quite significantly something like 70%). So, it appears to me that the area of the arc/circle functions IMHO needs some broader attempt to be resolved. E.g., one probably could argue that one more general arc function would be a better choice than the quarter arc function from Adafruits. This are not decision I feel I can make.

Reply:
You probably haven't even seen the code he's mentioning. Libopenui is a new UI library that is going to be introduced with 2.4 and replaces the current UI completely for color LCD radios.
Obviously this PR will have to be based on 2.4 to be accepted, so there you'll be able to make use of that.

Reply:
you are correct, haven't seen this. Not sure what conclusion I should draw from this now.

as regards when to merge, this certainly is exclusively your decision, and I wouldn't know if there would be a 2.3.9 or when 2.4 would hit the market, but I could imagine that there could be benefit from keeping it based on 2.3.x for some while, might be that there could be more testers that ways

Reply:
@olliw42 please have a look to libopenui function and check if the code there is better or not, if Adafruit's one is better then we take it also in libopenui, but I would like we have only one version.

Reply:
I'm happy to do so, but I will need some time for that.